extensions/amp-auto-ads/0.1/ad-strategy.js

```diff
@@ -28,25 +43,36 @@ export class AdStrategy {
    * @param {!Array<!DataAttributeDef>} baseDataAttributes Any data attributes
    *     that should be added to any inserted ads. These will be combined with
    *     any additional data atrributes specified by the configuration.
+   * @param {!./ad-tracker.AdTracker} adTracker
+   * @param {!StrategyConfigDef} strategyConfig
    */
-  constructor(type, placements, baseDataAttributes) {
+  constructor(type, placements, baseDataAttributes, adTracker, strategyConfig) {
     this.type_ = type;
 
     this.availablePlacements_ = placements.slice(0);
 
     this.baseDataAttributes_ = baseDataAttributes;
 
-    /** @type {!Array<!./placement.Placement>} */
-    this.placedAds_ = [];
+    /** @type {!./ad-tracker.AdTracker} */
+    this.adTracker_ = adTracker;
+
+    /** @type {!StrategyConfigDef} */
+    this.strategyConfig_ = strategyConfig;
   }
 
   /**
    * @return {!Promise<boolean>} True if strategy succeeds false otherwise.
    */
   run() {
-    // TODO: Implement a more extensive strategy based on parameters set in the
-    // configuration.
-    return this.placeNextAd_();
+    if (this.adTracker_.getAdCount() >= this.strategyConfig_.targetAdCount) {
+      return Promise.resolve(true);
+    }
+    return this.placeNextAd_().then(success => {
+      if (success) {
```

Comment:
Yes. The strategy now places multiple ads. So it keeps trying to place another ad until the target number of ads are on the page or it runs out of placements.